lib/state_machine/types.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "message.hpp"
+#include "state.hpp"
+
+namespace hyped::state_machine {
+
+struct Transition {
+  State from;
+  State to;
+  Message message;
```

Comment:
source, target, message?

something like `lookup(transition.to, transition.message)` doesn't read well, I think.

I am also not convinced we need this struct at all but I'll see.